inbox/sqlalchemy_ext/util.py

```diff
@@ -361,3 +362,24 @@ def safer_yield_per(query, id_field, start_id, count):
         results = query.filter(id_field >= cur_id).order_by(id_field).limit(count).all()
         yield from results
         cur_id = results[-1].id + 1
+
+
+mysql_dialect = mysql.dialect()
+
+
+def get_db_api_cursor_with_query(session, query):
+    """
+    Return a DB-API cursor with the given SQLAlchemy query executed.
+
+    This is useful when you want to optimize and skip the machinery of
+    SQLAlchemy ORM, particularly when you want to execute
+    a query that returns a large number of rows with a couple of columns.
+    SQLAlchemy ORM has to instantiate several Python objects for each row
+    returned by the query, which can be a performance bottleneck.
+    """
+    compiled_query = query.statement.compile(dialect=mysql_dialect)
```

Comment:
Could we use something like `session.get_bind().dialect` here to avoid hardcoding `mysql_dialect`?

Reply:
Done in [`d8fac30` (#957)](https://github.com/closeio/sync-engine/pull/957/commits/d8fac3081e9defd222b3befd39dbc39e59c3366e)